src/components/views/settings/tabs/user/SessionManagerTab.tsx

```diff
@@ -28,12 +30,60 @@ import Modal from '../../../../../Modal';
 import SetupEncryptionDialog from '../../../dialogs/security/SetupEncryptionDialog';
 import VerificationRequestDialog from '../../../dialogs/VerificationRequestDialog';
 import LogoutDialog from '../../../dialogs/LogoutDialog';
+import MatrixClientContext from '../../../../../contexts/MatrixClientContext';
+import { deleteDevicesWithInteractiveAuth } from '../../devices/deleteDevices';
+
+const useSignOut = (
+    matrixClient: MatrixClient,
+    refreshDevices: DevicesState['refreshDevices'],
+): {
+        onSignOutCurrentDevice: () => void;
+        onSignOutOtherDevices: (deviceIds: DeviceWithVerification['device_id'][]) => Promise<void>;
+        signingOutDeviceIds: DeviceWithVerification['device_id'][];
+    } => {
+    const [signingOutDeviceIds, setSigningOutDeviceIds] = useState<DeviceWithVerification['device_id'][]>([]);
+
+    const onSignOutCurrentDevice = () => {
+        Modal.createDialog(LogoutDialog,
+            /* props= */{}, /* className= */undefined,
```

Comment:
Uff. This is hard to read. Maybe we could improve this
- remove comments and hope people are using editors with param naming support or
- write each param on a line on its own and add the comment to the end `{}, // props`